### PR TITLE
Config cache

### PR DESCRIPTION
**Pull request by rucarrol-goog**
- Add tests for config.py
- remove voluptuous as dependency from config.py
- Add Config data class to model configuration.
- Create cached function to retrieve keys from configuration.

### CODE REVIEW

**Comment by awlx** [comment]
I like getting rid of dependencies which could potentially break the project. Also 👍 for the tests!

File: wgkex/config/config_test.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import unittest
+import mock
+import config
+import yaml
+
+_VALID_CFG = "domains:\n- a\n- b\nmqtt:\n  broker_port: 1883\n  broker_url: http://broker\n  keepalive: 5\n  password: pass\n  tls: true\n  username: user\npubkeys_file: some/file\n"
```
Comment (awlx) [inline wgkex/config/config_test.py:6]: I think we don't want the `http://` here as we are opening a `mqtt://` connection maybe use this as an URI string?

See https://github.com/mqtt/mqtt.org/wiki/URI-Scheme

**Review by awlx — CHANGES_REQUESTED** [review]

File: wgkex/config/config_test.py (hunk 1 above)
Comment (awlx) [inline wgkex/config/config_test.py:6]:
```suggestion
_VALID_CFG = "domains:\n- a\n- b\nmqtt:\n  broker_port: 1883\n  broker_url: mqtt://broker\n  keepalive: 5\n  password: pass\n  tls: true\n  username: user\npubkeys_file: some/file\n"
```

File: wgkex/config/config_test.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+_INVALID_LINT = "derpmains:\n- a\n- b\nmqtt:\n  broker_port: 1883\n  broker_url: http://broker\n  keepalive: 5\n  password: pass\n  tls: true\n  username: user\npubkeys_file: some/file\n"
```
Comment (awlx) [inline wgkex/config/config_test.py:7]:
```suggestion
_INVALID_LINT = "derpmains:\n- a\n- b\nmqtt:\n  broker_port: 1883\n  broker_url: mqtt://broker\n  keepalive: 5\n  password: pass\n  tls: true\n  username: user\npubkeys_file: some/file\n"
```

File: wgkex/config/config_test.py (hunk 1 above)
Comment (rucarrol-goog) [reply wgkex/config/config_test.py]: Makes sense! Thanks for pointing this out!

**Comment by rucarrol-goog** [comment]
> I like getting rid of dependencies which could potentially break the project. Also 👍 for the tests!

We still have 1 dependency for voluptuous left in `./broker/app.py`. I will not remove any dependency from requirements.txt until we have 100% coverage with `BUILD` files, and can produce a repeatable artifact.

**Comment by awlx** [comment]
Sounds good :).

**Pull request merged**